static/js/new-thing.js

```diff
@@ -232,6 +266,28 @@ class NewThing {
       this.pinInput.addEventListener('input', this.handlePinInput.bind(this));
       this.cancelButton.addEventListener('click', this.handleCancel.bind(this));
       this.submitButton.addEventListener('click', this.handleSubmit.bind(this));
+    } else if (this.requiresCredentials()) {
+      this.usernameInput =
+        this.element.getElementsByClassName('new-thing-username')[0];
```

Comment:
Complete nitpick, but this would be a bit cleaner as `this.element.querySelector('.new-thing-username')`

Reply:
Agreed. Done.